ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -493,10 +493,21 @@ public void syncWithSnapshot(long lastSnapshotIndex) {
 
     // Close open log segment if entries are already included in snapshot
     LogSegment openSegment = cache.getOpenSegment();
-    if (openSegment != null && openSegment.getEndIndex() <= lastSnapshotIndex) {
-      fileLogWorker.closeLogSegment(openSegment);
+    long purgeIndex = lastSnapshotIndex;
+    if (openSegment != null && openSegment.hasEntries()) {
+      LOG.debug("Found open segment {}, with end index {}, snapshotIndex {}",
```

Comment:
Can we add function name in the message?

Reply:
Done